nixnet/database/_dbc_signal_value_table.py

```diff
@@ -0,0 +1,103 @@
+
+import collections
+import six
+import typing  # NOQA: F401
+
+from nixnet import _funcs
+from nixnet import constants
+
+
+class DbcSignalValueTable(collections.Mapping):
+    """Collection for accessing a DBC signal value table."""
+
+    def __init__(self, handle):
+        # type: (int) -> None
+        self._handle = handle
+
+    def __repr__(self):
+        return '{}(handle={})'.format(type(self).__name__, self._handle)
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            return self._handle == typing.cast(DbcSignalValueTable, other)._handle
+        else:
+            return NotImplemented
+
+    def __ne__(self, other):
+        result = self.__eq__(other)
+        if result is NotImplemented:
+            return result
+        else:
+            return not result
+
+    def __hash__(self):
+        return hash(self._handle)
+
+    def __len__(self):
+        return len(self._value_table)
+
+    def __iter__(self):
+        return self.keys()
+
+    def __getitem__(self, key):
+        # type: (typing.Text) -> int
+        """Return the value.
+
+            Args:
+                Value description.
+            Returns:
+                Value
+        """
+        if isinstance(key, six.string_types):
+            return self._value_table[key]
+        else:
+            raise TypeError(key)
+
+    def keys(self):
+        """Return all value descriptions in the collection.
+
+            Yields:
+                An iterator to all value descriptions in the collection.
+        """
+        for key in self._value_table.keys():
+            yield key
+
+    def values(self):
+        """Return all values in the collection.
+
+            Yields:
+                An iterator to all values in the collection.
+        """
+        for value in self._value_table.values():
+            yield value
+
+    def items(self):
+        """Return all value descriptions and values in the collection.
+
+            Yields:
+                An iterator to tuple pairs of value descriptions and values in the collection.
+        """
+        for item in self._value_table.items():
+            yield item
+
+    @property
+    def _value_table(self):
+        # type: () -> typing.Dict[typing.Text, int]
+        mode = constants.GetDbcAttributeMode.VALUE_TABLE_LIST
+        attribute_size = _funcs.nxdb_get_dbc_attribute_size(self._handle, mode, '')
+        attribute_info = _funcs.nxdb_get_dbc_attribute(self._handle, mode, '', attribute_size)
+        table_string = attribute_info[0]
+
+        # table_string is of the format 'value1, key1, value2, key2, ...'
+        # convert to a dict: { 'key1': int('value1'), 'key2': int('value2'), ... }
+        table_list = [
+            element
+            for element in table_string.split(',')
+            if element.strip()]
+        table_dict = dict(
+            (key, int(value))
+            for value, key in zip(table_list[0::2], table_list[1::2]))
```

Comment:
`if element.strip()` 
Why are we ignoring empty strings when creating a list? Wouldn't this mess with the set of key-value pairings?

Also, why we casting value to int? I didn't see anything about value always being an int. I was thinking we just store the string representation as is (like we do for DBC attributes).

I'm thinking the code would look something like this:

table_list = table_string.split(',')
table_dict = dict(
  (key, value)
  for value, key in zip(table_list[0::2], table_list[1::2])
)

Reply:
"Why are we ignoring empty strings when creating a list? Wouldn't this mess with the set of key-value pairings?"

From my experience in playing with DBC creator, I was not concerned about empty keys. I opting for the case where I get `[]` instead of `['']` if `table_string == ''`. I'm happy to guard against empty keys though...I'll make the change.

> Also, why we casting value to int?

I'll send you information from a DBC creation tool that supports value being an int.

Reply:
Per our conversation, we'll hold off on making value an int unless I can be 100% sure that is all it will ever be.

Reply:
Fixed.